### PR TITLE
Updated Steam NSFW tagging.

### PR DESCRIPTION
**Pull request by Leon-JS02**
The steam extract script now looks for a `Mature Content Warning` when tagging games as NSFW.

### CODE REVIEW

File: web_scraping/steam_scraper/scrape_steam.py [hunk 1]
```diff
@@ -115,6 +115,12 @@ def scrape_game_tags(app_soup: bs4.BeautifulSoup) -> list[str]:
     return tags if tags else []
 
 
+def scrape_game_nsfw(app_soup: bs4.BeautifulSoup) -> bool:
+    """Returns True if a game is tagged as NSFW."""
+    headings = [heading.text.lower() for heading in app_soup.find_all('h2')]
+    return any('mature content' in heading for heading in headings)
```
Comment (mehrdadhalali) [inline web_scraping/steam_scraper/scrape_steam.py:121]: Neat trick!

File: web_scraping/steam_scraper/scrape_steam.py
Comment (mehrdadhalali) [inline web_scraping/steam_scraper/scrape_steam.py:1]: Technically speaking, we have no need for tags anymore, so at some point we can update our scripts to get rid of them.

**Review by mehrdadhalali — APPROVED** [review]
Nice NSFW detector, this should work perfectly.

File: web_scraping/steam_scraper/scrape_steam.py
Comment (Leon-JS02) [reply web_scraping/steam_scraper/scrape_steam.py:1]: While this is true, tags could still be used for future visualisations on the dashboard.

**Review by paulauus — APPROVED** [review]

**Review by MichaelHassan — APPROVED** [review]
I like how concise this is. Creamy stuff

**Pull request merged**